src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -106,15 +106,15 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
       } else {
         deleteFile(nameDir);
         nameDir.mkdirs();
-        runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
-        startProcess(driver, nameNodeTask);
-        startProcess(driver, zkfcNodeTask);
-        driver.sendStatusUpdate(TaskStatus.newBuilder()
-            .setTaskId(nameNodeTask.taskInfo.getTaskId())
-            .setState(TaskState.TASK_RUNNING)
-            .setMessage(messageStr)
-            .build());
       }
+      runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
+      startProcess(driver, nameNodeTask);
+      startProcess(driver, zkfcNodeTask);
+      driver.sendStatusUpdate(TaskStatus.newBuilder()
+          .setTaskId(nameNodeTask.taskInfo.getTaskId())
+          .setState(TaskState.TASK_RUNNING)
+          .setMessage(messageStr)
+          .build());
```

Comment:
This will create a large bug in our HA features so this is not mergable. Basically you are reformatting the NN when it's been already formatted. I'm guessing the issue you are encountering is when you want to redeploy HDFS and start from scratch without recovering data. In that case, I suggest you remove /tmp/hdfs data directories.